### PR TITLE
feat: Add Upload/Import configs to Slice/File entities

### PR DESCRIPTION
**Pull request by michaljurecko**
Jira: https://keboola.atlassian.net/browse/PSGO-598

**Changes:**
- Added `Slice.Staging.Upload`, `File.Staging.Upload`, `File.Target.Import` fields.
- These fields contains a snapshot of the upload/import configuration for the Coordinator node.

-----------

### CODE REVIEW

File: internal/pkg/service/stream/storage/level/staging/model/file.go [hunk 1]
```diff
@@ -20,6 +21,8 @@ type File struct {
 	Compression compression.Config `json:"compression" validate:"required"`
 	// Expiration determines how long it is possible to write to the staging file, e.g. due to expiring credentials.
 	Expiration utctime.UTCTime `json:"expiration" validate:"required"`
+	// Upload - slices upload configuration.
+	Upload config.UploadConfig `json:"upload"`
 }
```
Comment (michaljurecko) [inline internal/pkg/service/stream/storage/level/staging/model/file.go:26]: `File.Staging.Upload`

File: internal/pkg/service/stream/storage/level/staging/model/slice.go [hunk 2]
```diff
@@ -2,11 +2,14 @@ package model
...
 
 type Slice struct {
 	// Path to the slice in the staging File.
 	Path string `json:"path" validate:"required"`
 	// Compression configuration.
-	Compression compression.Config `json:"compression"  `
+	Compression compression.Config `json:"compression"`
+	// Upload - slice upload configuration.
+	Upload config.UploadConfig `json:"upload"`
 }
```
Comment (michaljurecko) [inline internal/pkg/service/stream/storage/level/staging/model/slice.go:15]: `Slice.Staging.Upload`

File: internal/pkg/service/stream/storage/level/target/model/target.go [hunk 3]
```diff
@@ -1,8 +1,12 @@
 package model
 
+import "github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/level/target/config"
+
 type Target struct {
 	// Provider of the target data destination.
 	Provider Provider `json:"provider" validate:"required"`
+	// Import - file import configuration.
+	Import config.ImportConfig `json:"import"`
 }
```
Comment (michaljurecko) [inline internal/pkg/service/stream/storage/level/target/model/target.go:10]: `File.Target.Import`

**Review by Matovidlo — APPROVED** [review]
LGTM

**Pull request merged**